public/app/features/alerting/unified/utils/matchers.ts

```diff
@@ -11,10 +18,18 @@ const matcherOperators = [
   MatcherOperator.equal,
 ];
 
+/**
+ * Parse a single matcher, examples:
+ *  foo="bar" => { name: foo, value: bar, isRegex: false, isEqual: true }
+ *  bar!~baz => { name: bar, value: baz, isRegex: true, isEqual: false }
+ */
 export function parseMatcher(matcher: string): Matcher {
   if (matcher.startsWith('{') && matcher.endsWith('}')) {
-    throw new Error(`PromQL matchers not supported yet, sorry! PromQL matcher found: ${matcher}`);
+    throw new Error(
```

Comment:
Is there a reason to keep this function alive? Shouldn't we migrate everything to `parseMatcherToArray`?

Reply:
There are a few code branches still calling this (for example to decode matchers via query string params for pre-populating matchers in the "new silence" page) since those don't support PromQL-style matchers.